### ISSUE TITLE
Durable function orchestration won't failed immediately until all outstanding tasks finished

### ISSUE THREAD

**Issue by Gorden22** [Azure/azure-functions-durable-extension#653]
**Describe the bug**
When one orchestration throw exception, durable function won't failed it immediately until all outstanding tasks finished or failed

**Investigative information**

- Durable Functions extension version:1.7.1
- Function App version (1.0 or 2.0):
- Programming language used: C#

***If JavaScript***

- durable-functions npm module version:

***If deployed to Azure***

- Timeframe issue observed:
- Function App name:
- Function name(s):
- Region:
- Orchestration instance ID(s):

> If you don't want to share your Function App name or Functions names on GitHub, please be sure to provide your Invocation ID, Timestamp, and Region - we can use this to look up your Function App/Function. Provide an invocation id per Function. See the [Functions Host wiki](https://github.com/Azure/azure-webjobs-sdk-script/wiki/Sharing-Your-Function-App-name-privately) for more details.

**To Reproduce**

> While not required, providing your orchestrator's source code in anonymized form is often very helpful when investigating unexpected orchestrator behavior.

**Expected behavior**

**Actual behavior**
A clear and concise description of what actually happened.

**Screenshots**

**Known workarounds**
Provide a description of any known workarounds you used.

**Additional context**

- Development environment (ex. Visual Studio)
- Links to source
- Additional bindings used
- Function invocation IDs

**Comment by cgillum** [comment]
This is by design in the Durable Task Framework, as shown by the code [here](https://github.com/Azure/durabletask/blob/master/src/DurableTask.Core/TaskOrchestrationExecutor.cs#L84). Is the ask to change this behavior?

**Comment by Gorden22** [comment]
> This is by design in the Durable Task Framework, as shown by the code here. Is the ask to change this behavior?

Hi Chris, so we think this is kind of not make sense to us. If task need to set to completed, it make sense all in flights tasks need finished (faulted/completed/cancelled), but if in the code we want terminate the orchestration, we throw exception. But at same time, if some task still running, actually users not care about them  this time, because if we want failed the workflow it means something wrong and orchestration should stop at this point, if  tasks is very long, it may make the orchestration just stuck there. Is this a point make sense? Please advise

**Comment by cgillum** [comment]
Yes, I understand your point and I think it makes sense. We will need to check with the DTFx maintainers to see if they are okay with changing this behavior.

**Comment by YunchuWang** [comment]
@cgillum, following up on your earlier comment asking whether this behavior should change: the current DTFx `TaskOrchestrationExecutor` still waits for `!context.HasOpenTasks` before turning a faulted orchestrator result into a failed orchestration. The same code also still contains a TODO for a setting that would allow completion when the orchestrator function completes even if durable tasks remain open:

https://github.com/Azure/durabletask/blob/main/src/DurableTask.Core/TaskOrchestrationExecutor.cs

So the requested alternative behavior has not been implemented, and any change appears to belong in `Azure/durabletask` rather than this extension. Would you prefer that we keep/transfer this enhancement to the DTFx repository, or close it here as an intentional design that we don't plan to change? I don't think "Completed" would accurately describe its current state.

**Comment by cgillum** [comment]
@YunchuWang thanks for bringing this up. You're correct that the change belongs in `Azure/durabletask`. However, rather than moving this issue into the DTFx repo, I suggest we keep this one here (as a "breadcrumb") and link it to a new issue in the DTFx issue. That way users of Durable Functions have better visibility into this issue. We can then close this issue if/when we close out the DTFx issue.

That said, I don't see this as a high-priority issue at this time. We may also be able to enforce this behavior in the newer Durable Task SDKs rather than in DTFx since the newer SDKs have their own embedded version of `TaskOrchestrationExecutor`, allowing them to implement different behavior without impacting existing DTFx workloads.
